packages/components/tsconfig.json

```diff
@@ -1,24 +1,26 @@
 {
+  // Manage references here
+  "references": [],
+
+  // Defaults (don't change this)
+  "extends": "../../tsconfig.packages.json",
   "compilerOptions": {
-    "experimentalDecorators": true,
-    "esModuleInterop": true,
+    "rootDir": "src",
     "outDir": "dist",
-    "module": "esnext",
-    "lib": ["dom", "es2017"],
-    "sourceMap": true,
+    "tsBuildInfoFile": "dist/.tsbuildinfo",
+
+    // Specific settings for npm package
+    "target": "es6",
+    "declaration": true,
+    "sourceMap": false,
+
+    // We use react
     "jsx": "react",
-    "moduleResolution": "node",
     "resolveJsonModule": true,
-    "target": "es5",
-    "skipLibCheck": false,
-    "allowSyntheticDefaultImports": true,
-    "strict": true,
-    "forceConsistentCasingInFileNames": true,
-    "isolatedModules": true,
-    "noImplicitAny": false,
-    "declaration": true,
+
+    // TODO: remove this
     "baseUrl": "./src"
```

Comment:
We don't use baseUrl in this repo. I put it there for now to keep the PR small. I'll make a different PR that removes this. Basically the reasoning behind this is that having a baseUrl (or any sort of paths config) creates multiple different ways of importing a source file. Some ways are preffered over others, and the IDE doesn't always suggest the best one, so we either waste time in PRs asking to import the file in the correct way, or we don't bother with that and the codebase becomes a mess of different preferences of how that or this programmer prefers to imports their files (see https://github.com/botpress/botpress).

You can see as well that the codebase doesn't make any use of `export default` and doesn't use any `index` files except for one index file at the root of every package to control the exports. The same thing could be done in the components package. There isn't any good reason in my opinion that files that imports a class should decide to name it to whatever it wants. Instead of going with the "ok I'll name the export from this file like this everywhere I import it" strategy we can just leverage the typescript compiler to check for us that we're importing the correct thing by using named exports.

Let me know what you think

Reply:
Solid argument, I agree, let's go that way!